src/main/io/serial.h

```diff
@@ -77,20 +77,26 @@ typedef enum {
     BAUD_COUNT
 } baudRate_e;
 
-extern const uint32_t baudRates[];
+extern const uint32_t baudRates[BAUD_COUNT];
 
 // serial port identifiers are now fixed, these values are used by MSP commands.
 typedef enum {
     SERIAL_PORT_ALL = -2,
     SERIAL_PORT_NONE = -1,
-    // prepare for transition to SERIAL_PORT_UART0
-    SERIAL_PORT_UART_FIRST = 0,
-#if SERIAL_UART_FIRST_INDEX == 0
+    SERIAL_PORT_LEGACY_START_IDENTIFIER = 0,
+    SERIAL_PORT_START_IDENTIFIER = 20,
+    SERIAL_PORT_USB_VCP = 20,
+
+    SERIAL_PORT_SOFTSERIAL_FIRST = 30,
+    SERIAL_PORT_SOFTSERIAL1 = SERIAL_PORT_SOFTSERIAL_FIRST,
+    SERIAL_PORT_SOFTSERIAL2,
+
+    SERIAL_PORT_LPUART_FIRST = 40,
+    SERIAL_PORT_LPUART1 = SERIAL_PORT_LPUART_FIRST,
+
+    SERIAL_PORT_UART_FIRST = 50,
```

Comment:
SERIAL_PORT_UART_FIRST should be 50(uart0) or 51(old targets). otherwise it will break saved resource mapping on existing targets..

Reply:
Adjusted.